gnomad_constraint/pipeline/constraint_pipeline.py

```diff
@@ -0,0 +1,124 @@
+        )
+
+    try:
+        if args.preprocess_data:
+            logger.info("Adding VEP context annotations...")
+            # Add annotations from VEP context Table to gnomAD data.
+            # TODO: Need to add function that annotates methylation, coverage, and gerp in the vep context table.
+            preprocessed_genome_ht = add_vep_context_annotations(
+                public_release("genomes").ht(),
+                annotated_context_ht.versions[version].ht(),
+            )
+            preprocessed_exome_ht = add_vep_context_annotations(
+                public_release("exomes").ht(),
+                annotated_context_ht.versions[version].ht(),
+            )
+            # Filter input Table and add annotations used in constraint calculations.
+            full_context_ht = prepare_ht_for_constraint_calculations(
+                annotated_context_ht.versions[version].ht()
+            )
+            full_genome_ht = prepare_ht_for_constraint_calculations(
+                preprocessed_genome_ht
+            )
+            full_exome_ht = prepare_ht_for_constraint_calculations(
+                preprocessed_exome_ht
+            )
+            # Filter to locus that is on an autosome or a pseudoautosomal region in context Table, exome Table, and genome Table.
+            context_ht = full_context_ht.filter(
+                full_context_ht.locus.in_autosome_or_par()
+            ).write(
+                get_preprocessed_ht("context", version).path, overwrite=args.overwrite
+            )
+            genome_ht = full_genome_ht.filter(
+                full_genome_ht.locus.in_autosome_or_par()
+            ).write(
+                get_preprocessed_ht("genome", version).path, overwrite=args.overwrite
+            )
+            exome_ht = full_exome_ht.filter(
+                full_exome_ht.locus.in_autosome_or_par()
+            ).write(
+                get_preprocessed_ht("exome", version).path, overwrite=args.overwrite
+            )
+            # Sex chromosomes are analyzed separately, since they are biologically different from the autosomes.
+            context_x_ht = filter_x_nonpar(full_context_ht).write(
+                get_preprocessed_ht("context", "chrx", version).path,
+                overwrite=args.overwrite,
+            )
+            context_y_ht = filter_y_nonpar(full_context_ht).write(
+                get_preprocessed_ht("context", "chry", version).path,
+                overwrite=args.overwrite,
+            )
+            exome_x_ht = filter_x_nonpar(full_exome_ht).write(
+                get_preprocessed_ht("exome", "chrx", version).path,
+                overwrite=args.overwrite,
+            )
+            exome_y_ht = filter_y_nonpar(full_exome_ht).write(
+                get_preprocessed_ht("exome", "chrx", version).path,
+                overwrite=args.overwrite,
+            )
+            logger.info("Done with preprocessing genome and exome Table.")
```

Comment:
```suggestion
            context_ht = annotated_context_ht.versions[version].ht()
            if test:
                rg = get_reference_genome(context_ht.locus)
                contigs_keep = [
                    hl.parse_locus_interval(c, reference_genome=rg)
                    for c in [rg.contigs[20], rg.x_contigs[0], rg.y_contigs[0]]
                ]
                context_ht = hl.filter_intervals(context_ht, contigs_keep)

            for data_type in DATA_TYPES:
                if data_type is "context":
                    preprocessed_ht = context_ht
                else:
                    ht = public_release(data_type).ht()
                    if test:
                        ht = hl.filter_intervals(ht, contigs_keep)
                    preprocessed_ht = add_vep_context_annotations(ht, context_ht)

                # Filter input Table and add annotations used in constraint calculations.
                full_ht = prepare_ht_for_constraint_calculations(preprocessed_ht)

                # Filter to locus that is on an autosome or in a pseudoautosomal region.
                ht = full_ht.filter(full_ht.locus.in_autosome_or_par())
                ht.write(get_preprocessed_ht(data_type, version, "autosome_par", test).path, overwrite=overwrite)

                # Sex chromosomes are analyzed separately, since they are biologically different from the autosomes.
                if data_type is not "genomes":
                    filter_x_nonpar(full_ht).write(
                        get_preprocessed_ht(data_type, version, "chrx_nonpar", test).path,
                        overwrite=overwrite,
                    )
                    filter_y_nonpar(full_ht).write(
                        get_preprocessed_ht(data_type, version, "chry_nonpar", test).path,
                        overwrite=overwrite,
                    )

            logger.info("Done with preprocessing genome and exome Table.")
```

Reply:
1. We did partition in the creating training dataset part, which has similar purpose. Do we want to do both or either of these? I feel we may want to choose the one that can immediately have the result, otherwise this option is useless.
2. Looks like the sex chromosomes aren't optional and skippable in the pipeline. I will ask in the channel tomorrow.

Reply:
1. Yes, sorry, my review for the other PR wasn't complete yet, but I had been changing that one to use the output test files that this creates. This does take a bit longer, but it was still pretty fast for testing when I tried it out, and it's certainly faster than the full dataset. If you want it faster you can change to do the partitioning, but we want to test the X and Y chromosomes as well. Also, the genome, exome, and context will all have different partitioning so this was the safer testing option.
2. I didn't skip them above for anything other than the genomes, and I don't think they are needed for that

Reply:
thanks for cleaning this section up @jkgoodrich! I like the structure for the processed hts

Reply:
also just noting here will need rg.contigs[20] -> rg.contigs[19]  to grab chromosome 20

Reply:
Great catch, thank you!